src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -187,21 +194,24 @@ ReportAie::writeReport(const xrt_core::device* /*_pDevice*/,
 
         if(tile.second.find("events") != tile.second.not_found()) {
           _output << boost::format("    %s:\n") % "Events";
-          for(auto& node : tile.second.get_child("events",empty_ptree)) {
+          for(const auto& node : tile.second.get_child("events", empty_ptree)) {
             _output << fmt8("%s")  % node.second.get<std::string>("name")
                                    % node.second.get<std::string>("value");
           }
           _output << std::endl;
         }
       }
-      _output << boost::format("    %s\n") % "Pl Kernel Instances in Graph:";
-      for (auto& node : graph.get_child("pl_kernel")) {
-        _output << boost::format("      %s\n") % node.second.data();
+
+      const boost::property_tree::ptree& pl_kernel = graph.get_child("pl_kernel");
+      if (!pl_kernel.empty()) {
+        _output << boost::format("    %s\n") % "Pl Kernel Instances in Graph:";

```

Comment:
I believe, the fix is required for all over ReportAie.cpp and ReportAieShim.cpp. Which is a big change and also need to verify all the outputs. 
I have created a separate CR for 2022.2 to fix this formatting issue. 
https://jira.xilinx.com/browse/CR-1125796

Reply:
Thank you for creating the CR.  And yes there is a lot of changes.  Good news is that I am not asking to do that over the entire code base, just for the code that was modified.  Cut-n-paste bad code only causes more bad code to propagate, hence my request to clean up this "new" code.

That said, being that we are at the _end of the release_ and that you have created a CR, I am O.K. to overlook this issue.  Especially since, make it now could introduce a bug / issue.